snappy-sys/build.rs

```diff
@@ -45,61 +38,37 @@ fn configure_snappy(want_static: bool) -> bool {
 }
 
 fn build_snappy() {
-    let src = PathBuf::from(&env::var("CARGO_MANIFEST_DIR").unwrap()).join("snappy");
-    let build = PathBuf::from(&env::var("OUT_DIR").unwrap()).join("build");
+    let out_dir = PathBuf::from(&env::var("OUT_DIR").unwrap());
 
-    let cc = gcc::Config::new().get_compiler();
+    let cc = cc::Build::new().get_compiler();
     let mut cflags = OsString::new();
     for arg in cc.args() {
         cflags.push(arg);
         cflags.push(" ");
     }
-    t!(fs::create_dir_all(&build));
-    cp_r(&src, &build);
-    run(Command::new("./autogen.sh").current_dir(&build));
-    run(Command::new("./configure")
+    cmake::Config::new("snappy")
         .env("CC", cc.path())
         .env("CFLAGS", cflags)
-        .current_dir(&build)
-        .arg("--disable-shared")
-        .arg("--enable-static")
-        .arg("--with-pic"));
-    run(Command::new("make").current_dir(&build));
+        .env("CMAKE_BUILD_TYPE", "static")
+        .build();
     println!("cargo:rustc-link-lib=static=snappy");
-    println!("cargo:rustc-link-search={}/.libs", build.to_string_lossy());
-    println!("cargo:root={}", build.to_string_lossy());
+    println!("cargo:rustc-link-search={}", out_dir.join("lib").to_string_lossy());
+    println!("cargo:root={}", out_dir.to_string_lossy());
     configure_stdcpp();
 }
 
 fn configure_stdcpp() {
-    // From: https://github.com/alexcrichton/gcc-rs/blob/master/src/lib.rs
+    // From: https://github.com/alexcrichton/cc-rs/blob/master/src/lib.rs
     let target = env::var("TARGET").unwrap();
-    let cpp = if target.contains("darwin") { "c++" } else { "stdc++" };
-    println!("cargo:rustc-link-lib={}", cpp);
-}
-
-fn cp_r(dir: &Path, dst: &Path) {
-    for entry in t!(fs::read_dir(dir)) {
-        let entry = entry.expect("entry");
-        let path = entry.path();
-        let dst = dst.join(path.file_name().unwrap());
-        if t!(fs::metadata(&path)).is_file() {
-            t!(fs::copy(path, dst));
-        } else {
-            t!(fs::create_dir_all(&dst));
-            cp_r(&path, &dst);
-        }
-    }
-}
-
-fn run(cmd: &mut Command) {
-    println!("running: {:?}", cmd);
-    let status = match cmd.status() {
-        Ok(s) => s,
-        Err(e) => panic!("failed to run: {}", e),
+    let cpp = if target.contains("darwin") {
+        Some("c++")
+    } else if target.contains("windows") {
+        None
+    } else {
+        Some("stdc++")
     };
-    if !status.success() {
-        panic!("failed to run successfully: {}", status);
+    if let Some(cpp) = cpp {
+        println!("cargo:rustc-link-lib={}", cpp);
```

Comment:
It looks to me like the correct values can be retrieved from [the cc crate](https://github.com/alexcrichton/cc-rs/blob/master/src/lib.rs#L1973). And depending on how you are using `Build` it will also [configure cargo](https://github.com/alexcrichton/cc-rs/blob/master/src/lib.rs#L927).